packages/public/server/__tests-e2e__/login.ts

```diff
@@ -19,49 +19,122 @@
+  const loginPage = await elements
+    .getLoginButton(firstPage)
+    .then(clickForNewPage)
 
-      expect(firstPage).toHaveUrlPath(startPath)
+  expect(loginPage).toHaveUrlPath(pages.login.path)
 
-      const loginPage = await elements
-        .getLoginButton(firstPage)
-        .then(clickForNewPage)
+  const { buttonLogin, inputUser, inputPassword } = pages.login.identifier
+  await typeIntoLoginForm('123456', loginPage, 'login')
 
-      expect(loginPage).toHaveUrlPath(pages.login.path)
+  const afterLoginPage = await getByText(loginPage, buttonLogin).then(
+    clickForNewPage
+  )
 
-      const { buttonLogin, inputUser, inputPassword } = pages.login.identifier
-      await getByPlaceholderText(loginPage, inputUser).then((e) => e.type(user))
-      await getByPlaceholderText(loginPage, inputPassword).then((e) =>
-        e.type(pages.login.defaultPassword)
+  await expect(await elements.getLogoutButton(afterLoginPage)).toBeDefined()
+
+  const userPage = await elements
+    .getProfileButton(afterLoginPage)
+    .then(clickForNewPage)
+
+  expect(userPage).toHaveUrlPath('/user/me')
+  await expect(userPage).toMatchElement('h1', { text: 'login' })
+})
+
+describe('wrong login-data', () => {
+  const { buttonLogin, inputUser, inputPassword } = pages.login.identifier
+  let loginPage: ElementHandle
+
+  beforeEach(async () => {
+    await page.setViewport(viewports.desktop)
+    loginPage = await goto('/auth/login')
+  })
+
+  test('wrong password and correct user', async () => {
+    await typeIntoLoginForm('123', loginPage, 'login')
+
+    const afterLoginPage = await getByText(loginPage, buttonLogin).then(
+      clickForNewPage
+    )
+
+    expect(afterLoginPage).toHaveUrlPath(pages.login.path)
+
+    expect(
+      await getByText(
+        afterLoginPage,
+        'Mit dieser Kombination ist bei uns kein Benutzer registriert.'
       )
+    ).toBeDefined
+  })
+
+  test('correct password and wrong user', async () => {
+    await typeIntoLoginForm('123456', loginPage, 'abc')
+    const afterLoginPage = await getByText(loginPage, buttonLogin).then(
+      clickForNewPage
+    )
+    expect(afterLoginPage).toHaveUrlPath(pages.login.path)
 
-      const afterLoginPage = await getByText(loginPage, buttonLogin).then(
-        clickForNewPage
+    expect(
+      await getByText(
+        afterLoginPage,
+        'Mit dieser Kombination ist bei uns kein Benutzer registriert.'
       )
+    ).toBeDefined
+  })
+})
 
-      await expect(await elements.getLogoutButton(afterLoginPage)).toBeDefined()
+describe('missing login-data', () => {
+  let loginPage: ElementHandle
 
-      const userPage = await elements
-        .getProfileButton(afterLoginPage)
-        .then(clickForNewPage)
+  beforeAll(async () => {
+    await page.setViewport(viewports.desktop)
+    loginPage = await goto('/auth/login')
+  })
+  test('missing password and missing user', async () => {
+    typeIntoLoginForm('', loginPage, '')
+    clickOnLoginButton(loginPage)
```

Comment:
```suggestion
    await typeIntoLoginForm('', loginPage, '')
    await clickOnLoginButton(loginPage)
```

Reply:
Result of both functions is a promise (those are async commands) -> you need to have an await to wait for the action to be done or otherwise the tests will stop them before completion.